src/Redux/Reducers/authSlice.js

```diff
@@ -0,0 +1,111 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import { AlertToast, SuccessToast } from "Components";
+import axios from "axios";
+
+const encodedToken = localStorage.getItem("StormMediaToken");
+const initialState = {
+  currentUser: {},
+};
+
+export const login = createAsyncThunk("auth/login", async (userDetails) => {
+  try {
+    const response = await axios.post(`/api/auth/login`, {
+      username: userDetails.username,
+      password: userDetails.password,
```

Comment:
Suggestion: if you spread user details, you can avoid creating an object again

Reply:
Thanks for suggestion will look into it